internal/conv/conv_test.go

```diff
@@ -236,3 +236,125 @@ func TestF32stos(t *testing.T) {
 		})
 	}
 }
+
+func TestUtf8ToSjis(t *testing.T) {
+	type args struct {
+		s string
+	}
+	type want struct {
+		want string
+		err  error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, string, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got string, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		if !reflect.DeepEqual(got, w.want) {
+			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return sjis string from UTF8 string",
+			args: args{
+				s: "こんにちは",
+			},
+			want: want{
+				want: "\x82\xb1\x82\xf1\x82ɂ\xbf\x82\xcd",
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		test := tc
+		t.Run(test.name, func(tt *testing.T) {
+			tt.Parallel()
+			defer goleak.VerifyNone(tt, goleak.IgnoreCurrent())
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			checkFunc := test.checkFunc
+			if test.checkFunc == nil {
+				checkFunc = defaultCheckFunc
+			}
+
+			got, err := Utf8ToSjis(test.args.s)
+			if err := checkFunc(test.want, got, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestUtf8ToEucjp(t *testing.T) {
+	type args struct {
+		s string
+	}
+	type want struct {
+		want string
+		err  error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, string, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got string, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		if !reflect.DeepEqual(got, w.want) {
+			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return eucjp string from UTF8 string",
+			args: args{
+				s: "こんにちは",
+			},
+			want: want{
+				want: "\xa4\xb3\xa4\xf3\xa4ˤ\xc1\xa4\xcf",
+			},
+		},
```

Comment:
sorry I couldn't think of any error case for this test, do you have any idea?

Reply:
This is just idea, so it may enable for test to split function like below and to use `iotest.ErrReader` in `testing/iotest` package.
```
func utf8ToSjis(r io.Reader) (string, error) {
	b, err := io.ReadAll(transform.NewReader(r, japanese.ShiftJIS.NewEncoder()))
	if err != nil {
		return "", err
	}
	return string(b), nil
}

// Utf8ToSjis converts a UTF8 string to sjis string.
func Utf8ToSjis(s string) (string, error) {
	return utf8ToSjis(strings.NewReader(s))
}
```

Reply:
fixed by spliting the ecode()
changes: https://github.com/vdaas/vald/pull/1678/commits/5ad9c5ab43c62665832eb9e8002ce5697f211104